src/linter/block.go

```diff
@@ -2302,7 +2309,19 @@ func (b *BlockWalker) handleAssign(a *assign.Assign) bool {
 		b.checkVoidType(a.Expression)
 		b.replaceVar(v, solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, a.Expression), "assign", meta.VarAlwaysDefined)
 	case *expr.List:
-		b.handleAssignList(v.Items)
+		if !meta.IsIndexingComplete() {
+			return true
+		}
+
+		tp := solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, a.Expression)
+		var findType string
```

Comment:
`findType` is a verb. It's not a good variable name here in my opinion.
You can call it `shapeType` given that it's only assigned to shape types.